vsts/config/utils/browsers.js

```diff
@@ -4,26 +4,98 @@
 const logger = require('@blackbaud/skyux-logger');
 const get = require('lodash.get');
 
+const bsBrowserChrome = {
+  os: 'Windows',
+  osVersion: '10',
+  browser: 'Chrome'
+};
+
+const bsBrowserEdge = {
+  os: 'Windows',
+  osVersion: '10',
+  browser: 'Edge'
+};
+
+const bsBrowserIE = {
+  os: 'Windows',
+  osVersion: '7',
+  browser: 'IE',
+  browserVersion: '11'
+};
+
+const bsBrowserFirefox = {
+  os: 'OS X',
+  osVersion: 'High Sierra',
+  browser: 'Firefox'
+};
+
+const bsBrowserSafari = {
+  os: 'OS X',
+  osVersion: 'High Sierra',
+  browser: 'Safari'
+};
+
+const browserSets = {
+  speedy: [
+    bsBrowserChrome
+  ],
+  quirky: [
+    bsBrowserChrome,
+    bsBrowserEdge,
+    bsBrowserIE
+  ],
+  paranoid: [
+    bsBrowserChrome,
+    bsBrowserEdge,
+    bsBrowserIE,
+    bsBrowserFirefox,
+    bsBrowserSafari
+  ]
+};
+
 // We normalize properties despite Browserstack/Protractor/Karma using different keys.
-const map = {
-  'e2e': {
+// Leaving this functionaly in case we expose the e2e property in the future.
+const propertiesMap = {
+  e2e: {
     osVersion: 'os_version',
     browser: 'browserName',
     browserVersion: 'browser_version',
   },
-  'unit': {
+  unit: {
     osVersion: 'os_version',
     browserVersion: 'browser_version',
   }
 };
 
+function validateBrowserSet(configBrowserSet) {
+  if (configBrowserSet) {
+    logger.info(`Attempting to use config setting ${configBrowserSet}.`);
+    const validBrowserSetKeys = Object.keys(browserSets);
+    const browserSetKeyIndex = validBrowserSetKeys.indexOf(configBrowserSet.toLowerCase());
+
+    if (browserSetKeyIndex > -1) {
+      logger.info(`Validated config setting ${configBrowserSet}.`);
+      return validBrowserSetKeys[browserSetKeyIndex];
+    }
+  }
+}
+
 module.exports = {
   getBrowsers: (config, testSuite, defaults) => {
 
-    const browsers = get(config, `skyPagesConfig.skyux.testSettings.${testSuite}.browsers`, []);
-    const allowedPropertiesMap = map[testSuite];
+    const configBrowserSetRequsted = get(
+      config,
+      `skyPagesConfig.skyux.testSettings.${testSuite}.browserSet`,
+      ''
+    );
+    const configBrowserSetValidated = validateBrowserSet(configBrowserSetRequsted);
+
+    const allowedPropertiesMap = propertiesMap[testSuite];
     const allowedPropertiesKeys = Object.keys(allowedPropertiesMap);
 
+    // User hasn't defined a browserSet or provided an invalid set.
```

Comment:
Should be "User hasn't defined a browserSet, or has provided an invalid set."?

Also, is there a default browser set if the user does not provide one?

Reply:
I cleaned up the wording slightly, but I'm not quite following what was wrong with the original.  Let me know what you think about the new one.

Great question regarding a default browser.  If the property isn't set, or is set to something invalid, the local browser still runs.  Setting this property kicks you in to browserstack mode.

Reply:
Makes total sense, thanks for the clarification. In regard to the comment, I was reading it as:
"User hasn't defined a browserSet or (hasn't) provided an invalid set." (Basically, carrying the negative "hasn't" to the next section of the sentence.) It's not a huge deal either way :)

Reply:
Ahhh I see.  Yep that's confusing.  I'll make the comment more clear.